### PR TITLE
Network settings

### PR DESCRIPTION
**Pull request by BobrImperator**
- adds http#post method, but it's split into #56 

- adds Change network name form
it redirects to network-settings if change has succeeded
- adds Change network password form
it has some basic validation where new password and repeated field must have the same value
it redirects to network-settings if change has succeeded

### CODE REVIEW

**Review by marcoow — COMMENTED** [review]
I think there's one step missing here functionally which is the success screen. We'll need that since after changing either the network name or password, you'll need to reconnect (since the Wifi settings used previously are no longer valid). In the design we have the [success screen](https://share.goabstract.com/e224b8d6-8117-4698-b54c-1699bad6526b?collectionLayerId=1829798b-a640-4581-85ca-9066a0aec255&mode=design) for that but that doesn't have any navigational elements so not sure how one would proceed from there?

@marhigh can you clarify?

File: tests/acceptance/network-settings-test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  test("it has a correct description", async function (assert) {
+    await authenticateSession();
+    const currentDeviceService = this.owner.lookup("service:currentDevice");
+    currentDeviceService.data = { name: "Dummy" };
+    await visit("/network-settings");
+
+    assert
+      .dom("[data-test-layout-description]")
+      .hasText("Edit your Dummy mesh network settings");
+  });
```
Comment (marcoow) [inline tests/acceptance/network-settings-test.js:47]: I wouldn't add tests like this – they'd have to be changed every time the translations are changed…

Comment (BobrImperator) [reply tests/acceptance/network-settings-test.js:47]: Yeah, probably that's not necessary anymore since there's a linter for missing translations.

**Comment by marhigh** [comment]
> I think there's one step missing here functionally which is the success screen. We'll need that since after changing either the network name or password, you'll need to reconnect (since the Wifi settings used previously are no longer valid). In the design we have the [success screen](https://share.goabstract.com/e224b8d6-8117-4698-b54c-1699bad6526b?collectionLayerId=1829798b-a640-4581-85ca-9066a0aec255&mode=design) for that but that doesn't have any navigational elements so not sure how one would proceed from there?
> 
> @marhigh can you clarify?

I've added a button to return to the dashboard on the success view, please take a look: https://share.goabstract.com/e224b8d6-8117-4698-b54c-1699bad6526b?collectionLayerId=1829798b-a640-4581-85ca-9066a0aec255&mode=design

**Review by marcoow — APPROVED** [review]

**Pull request merged**